src/measure_theory/function/strongly_measurable.lean

```diff
@@ -0,0 +1,328 @@
+  let f_approx := hf_meas.approx,
+  let fs := λ n, simple_func.restrict (f_approx n) (S n ∩ t),
+  have h_fs_t_compl : ∀ n, ∀ x ∉ t, fs n x = 0,
+  { intros n x hxt,
+    rw simple_func.restrict_apply _ ((hS_meas n).inter ht),
+    refine set.indicator_of_not_mem _ _,
+    simp [hxt], },
+  refine ⟨fs, _, λ x, _⟩,
+  { simp_rw simple_func.support_eq,
+    refine λ n, (measure_bUnion_finset_le _ _).trans_lt _,
+    refine ennreal.sum_lt_top_iff.mpr (λ y hy, _),
+    rw simple_func.restrict_preimage_singleton _ ((hS_meas n).inter ht),
+    swap, { rw finset.mem_filter at hy, exact hy.2, },
+    refine (measure_mono (set.inter_subset_left _ _)).trans_lt _,
+    have h_lt_top := measure_spanning_sets_lt_top (μ.restrict t) n,
+    rwa measure.restrict_apply' ht at h_lt_top, },
+  { by_cases hxt : x ∈ t,
+    swap, { rw [funext (λ n, h_fs_t_compl n x hxt), hft_zero x hxt], exact tendsto_const_nhds, },
+    have h : tendsto (λ n, (f_approx n) x) at_top (𝓝 (f x)), from hf_meas.tendsto_approx x,
+    obtain ⟨n₁, hn₁⟩ : ∃ n, ∀ m, n ≤ m → fs m x = f_approx m x,
+    { obtain ⟨n, hn⟩ : ∃ n, ∀ m, n ≤ m → x ∈ S m ∩ t,
+      { suffices : ∃ n, ∀ m, n ≤ m → x ∈ S m,
+        { obtain ⟨n, hn⟩ := this,
+          exact ⟨n, λ m hnm, set.mem_inter (hn m hnm) hxt⟩, },
+        suffices : ∃ n, x ∈ S n,
+        { rcases this with ⟨n, hn⟩,
+          exact ⟨n, λ m hnm, monotone_spanning_sets (μ.restrict t) hnm hn⟩, },
+        rw [← set.mem_Union, Union_spanning_sets (μ.restrict t)],
+        trivial, },
+      refine ⟨n, λ m hnm, _⟩,
+      simp_rw [fs, simple_func.restrict_apply _ ((hS_meas m).inter ht),
+        set.indicator_of_mem (hn m hnm)], },
+    rw tendsto_at_top' at h ⊢,
+    intros s hs,
+    obtain ⟨n₂, hn₂⟩ := h s hs,
+    refine ⟨max n₁ n₂, λ m hm, _⟩,
+    rw hn₁ m ((le_max_left _ _).trans hm.le),
+    exact hn₂ m ((le_max_right _ _).trans hm.le), },
+end
+
+/-- If the measure is sigma-finite, all strongly measurable functions are
+  `fin_strongly_measurable`. -/
+protected lemma fin_strongly_measurable [topological_space β] [has_zero β] {m0 : measurable_space α}
+  (hf : strongly_measurable f) (μ : measure α) [sigma_finite μ] :
+  fin_strongly_measurable f μ :=
+hf.fin_strongly_measurable_of_exists_set_sigma_finite
+  ⟨set.univ, measurable_set.univ, by simp, by rwa measure.restrict_univ⟩
+
+/-- A strongly measurable function is measurable. -/
+protected lemma measurable [measurable_space α] [metric_space β] [measurable_space β]
+  [borel_space β] (hf : strongly_measurable f) :
+  measurable f :=
+measurable_of_tendsto_metric (λ n, (hf.approx n).measurable) (tendsto_pi.mpr hf.tendsto_approx)
+
+end strongly_measurable
+
+section second_countable_strongly_measurable
+variables {α β : Type*} [measurable_space α] [measurable_space β] {f : α → β}
+
+/-- In a space with second countable topology, measurable implies strongly measurable.
+  TODO: remove the `nonempty β` hypothesis? -/
```

Comment:
yes, please remove it :-) (if beta is empty, so is alpha, and you can take as approximating sequence the trivial sequence mapping nothing to nothing -- you could even state a lemma saying that, when the target space is empty, every function is strongly measurable).